src/remoteConnector.ts

```diff
@@ -810,12 +826,21 @@ export default class RemoteConnector extends Disposable {
 						gitpodHost: connectionInfo.gitpodHost
 					});
 				}
-				await this.context.globalState.update(remoteUri.authority, undefined);
+
+				// Don't await this on purpose so it doesn't block extension activation
+				this.startHeartBeat(connectionInfo);
```

Comment:
We should do it as soon as possible avoid any unnecessary async requests and so on. It should not be matter whether we connected or remote extensio is running. Only that a window is opened. We should try to emualte logic from here:
https://github.com/gitpod-io/gitpod/blob/663e22a5501669080979387c96d0196e91df2834/components/supervisor/frontend/src/index.ts#L230-L241

Reply:
This should be async because it has to call getSession and for that the extension needs to be initialized. This happens because we have both gitpod auth and also require a session in the same extension.